### PR TITLE
OPSEXP-590 Cleanup and uninstallation playbook

### PR DESCRIPTION
**Pull request by slohe1**
Cleanup and Uninstallation playbook added along with updated documentation.

### CODE REVIEW

File: docs/deployment-guide.md [hunk 1]
```diff
@@ -708,11 +708,21 @@ In that case, you can add the `cluster_keepoff` variable to one of the `reposito
...
-## Cleanup
+## Cleanup & Uninstallation of ACS 
 
-What needs to be removed from a system will depend on your inventory configuration. The steps below presume a localhost or single machine deployment i.e. where all roles were run on the same machine.
+What needs to be removed from a system will depend on your inventory configuration. The steps below presume a cleanup and uninstallation of Alfresco content service after deployment of ansible artifacts by using platform-cleanup.yml playbook and platform-uninstall.yml playbook respectively. 
 
-1. Stop and remove the following systemd services:
+# Cleanup
+
+This playbook will remove the temporary artifacts which are stored on the hosts.In order to cleanup the system post deployment run the following command:
```
Comment (alxgomz) [inline docs/deployment-guide.md:717]: Can you add a reminder about the fact running that playbook breaks playbook idempotency?

File: docs/deployment-guide.md [hunk 2]
```diff
@@ -708,11 +708,21 @@ In that case, you can add the `cluster_keepoff` variable to one of the `reposito
...
+
+ansible-playbook playbooks/platform-cleanup.yml -i inventory_ssh.yml
+
+# Uninstallation
+
+This playbook will uninstall the sevices belongs to the specific hosts. Below are the services, packages & folders we are removing as the uninstalltions
```
Comment (alxgomz) [inline docs/deployment-guide.md:723]:
```suggestion
This playbook will uninstall the sevices which belong to the specific hosts. Below are the services, packages & folders we are removing when uninstalling 
```

File: playbooks/platform-uninstall.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
+---
+#### This playbook will uninstall ACS
+
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:3]: Please avoid white space at the beginning of the file

File: playbooks/platform-uninstall.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ - hosts: database
+   tasks:
+     - name: Stopping the database
+       ansible.builtin.service:
+         name: postgresql{% if ansible_os_family != 'Debian' %}{{ dependencies_version.postgres_major_version }}{% endif %}
+         state: stopped
+         enabled: no
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:10]: unnecessary as we remove the package later on and the service is deployed with the package

File: playbooks/platform-uninstall.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+     - name: Removing the Packages   
+       ansible.builtin.package:
+         name: postgresql
+         state: absent
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:15]: Please use  `autoremove: yes` (`ansible.builtin.package` is just a wrapper for apt, dnf & yum  modules, so every option that's valid for all package managers is valid.

File: playbooks/platform-uninstall.yml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+ 
+ - hosts: activemq
+   tasks:
+     - name: Stopping the activemq
+       ansible.builtin.service:
+         name: activemq.service 
+         state: stopped
+         enabled: no
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:23]: Do not forget to remove the systemd unit file as well. This one is not installed by a package so needs to be removed  by the playbook (and systemd reloaded using an handler)

File: playbooks/platform-uninstall.yml [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+     - name: Removing the Packages   
+       ansible.builtin.package:
+         name: "{{ item }}"
+         state: absent
+       with_items:
+        - ImageMagick
+        - libreoffice
+
+     - name: Removing activemq folders
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:53]:
```suggestion
     - name: Removing Java folders
```

File: playbooks/platform-uninstall.yml [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+     - name: Stopping transform
+       ansible.builtin.service:
+         name: "{{ item }}" 
+         state: stopped
+         enabled: no
+       with_items:
+        - 'alfresco-transform-router.service'
+        - 'alfresco-tengine-aio.service'
+        - 'alfresco-shared-fs.service'
+
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:44]: Also. do not forget to cleanup the systemd unit files and reload daemon

File: playbooks/platform-uninstall.yml [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+ - hosts: search
+   tasks:
+     - name: Stopping Search
+       ansible.builtin.service:
+         name: alfresco-search.service 
+         state: stopped
+         enabled: no
+
+     - name: Removing activemq folders
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:68]: same as above. this is Java not activemq

File: playbooks/platform-uninstall.yml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+       with_items:
+        - "{{ java_home | default('/opt/openjdk-{{ dependencies_version.java }}') }}"
+
+ - hosts: search
+   tasks:
+     - name: Stopping Search
+       ansible.builtin.service:
+         name: alfresco-search.service 
+         state: stopped
+         enabled: no
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:66]: same as above

File: playbooks/platform-uninstall.yml [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+       with_items:
+        - "{{ java_home | default('/opt/openjdk-{{ dependencies_version.java }}') }}"
+
+ - hosts: syncservice
+   tasks:
+     - name: Stopping Sync
+       ansible.builtin.service:
+         name: alfresco-sync.service
+         state: stopped
+         enabled: no
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:81]: same as above.

File: playbooks/platform-uninstall.yml [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+     - name: Removing activemq folders
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:83]: java not activemq

File: playbooks/platform-uninstall.yml [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+       with_items:
+        - "{{ java_home | default('/opt/openjdk-{{ dependencies_version.java }}') }}"
+ 
+ - hosts: nginx
+   tasks:
+     - name: Stopping nginx
+       ansible.builtin.service:
+         name: nginx.service
+         state: stopped
+         enabled: no
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:96]: we don't need that as we remove the pkg

File: playbooks/platform-uninstall.yml [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+         path: "{{ item }}" 
+         state: absent
+       with_items:
+        - "{{ tomcat_home_prefix | default('/opt/apache-tomcat') }}"
+        - "{{ java_home | default('/opt/openjdk-{{ dependencies_version.java }}') }}"
+        - "{{ binaries_folder | default('/opt/alfresco') }}"
+        - "{{ config_folder | default('/etc/opt/alfresco') }}"
+        - "{{ data_folder | default('/var/opt/alfresco') }}"
+        - "{{ logs_folder | default('/var/log/alfresco') }}"
+        - "{{ download_location | default('/tmp/ansible_artefacts') }}"
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:122]: these one are not repository specific. Also transformers, search, syncservice have these folders

Comment (alxgomz) [reply playbooks/platform-uninstall.yml:122]: actually `{{ config_folder }}` might even be required elsewhere

File: playbooks/platform-uninstall.yml [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+         name: alfresco-content.service
+         state: stopped
+         enabled: no
+
+     - name: Removing the folders
+       ansible.builtin.file: 
+         path: "{{ item }}" 
+         state: absent
+       with_items:
+        - "{{ tomcat_home_prefix | default('/opt/apache-tomcat') }}"
```
Comment (alxgomz) [inline playbooks/platform-uninstall.yml:128]: missing the unit file

**Review by alxgomz — CHANGES_REQUESTED** [review]
please address remaining comment.
In particular take a look at the doc for ansible handlers and use it to reload systemd after a unit file is removed

**Review by alxgomz — CHANGES_REQUESTED** [review]
Please DRY. Use include statements in each play and have the handler defined only once in a separate file (e.g. `uninstall-handlers.yml`).
Also the pre-commi stage is still failing, please make sure to run pre-commit before pushing

**Review by gionn — CHANGES_REQUESTED** [review]
we should run this playbook as part of a molecule run, maybe we can hook it in the [cleanup](https://github.com/Alfresco/alfresco-ansible-deployment/blob/master/molecule/default/cleanup.yml) phase?

**Review by gionn — CHANGES_REQUESTED** [review]
see https://github.com/Alfresco/alfresco-ansible-deployment/runs/7561678936?check_suite_focus=true#step:11:127

**Comment by slohe1** [comment]
> see https://github.com/Alfresco/alfresco-ansible-deployment/runs/7561678936?check_suite_focus=true#step:11:127

**Comment by slohe1** [comment]
https://github.com/Alfresco/alfresco-ansible-deployment/pull/388#pullrequestreview-1055130 
Here looks like TASK [Removing the Packages]  passed. Stopping the playbook at next play. Just trying to get why it is different for this instance.

**Review by gionn — APPROVED** [review]

**Pull request merged**